### ISSUE TITLE
Jupyter - Beginning / Overview

### ISSUE THREAD

**Issue by danscime** [JacquesCarette/Drasil#1723]
Based on some new assignments from @JacquesCarette to see if there is a suitable way to implement Jupyter with Drasil.

It seemed like putting this under version control would be beneficial.
[Jupyter Branch](https://github.com/JacquesCarette/Drasil/tree/jupInit/People/DanScime/Jupyter-Tests)

I started by working through some cells to make a sample Jupyter page based on GamePhysics. Jupyter uses markdown in its cells, which can also render HTML.

Before trying to come up with a printer or anything, I tested more of the capability of Jupyter rendering HTML. I wrote a [little Haskell script ](https://github.com/JacquesCarette/Drasil/blob/jupInit/People/DanScime/Jupyter-Tests/Format%20script.hs) that removes the newlines from a file and changes quotation marks to have an escape character in front.

Then I took this new One Line html code and threw it into a cell and compiled the Jupyter notebook. So this is the current [Viewable Rendering of GamePhysics in Jupyter](https://github.com/JacquesCarette/Drasil/blob/jupInit/People/DanScime/Jupyter-Tests/GamePhysics-OneLine-Test.pdf). 

Some things (mainly the tables) have weird formatting that I will look into. And one main issue would be that right now, in the Jupyter page, the entire document shows up in one cell, which will make it hard to edit.

Depending on what the plans are for Jupyter, these may want to be fixed. It's possible that one giant cell may be fine, and then new cells underneath can be added for displaying runnable code.

**Comment by danscime** [comment]
Next steps are to try to build the Projectile example in Jupyter along with some runnable code. First, manually, then begin creating a printer.

**Comment by smiths** [comment]
Sounds good @danscime.  The first version of your Jupyter generated code can just be the documentation (SRS) in json format.  We can worry about embedding executable code in a subsequent step.

**Comment by danscime** [comment]
As of 865747a06e6032850133e467923810efc2d301cf, Projectile SRS is built in Jupyter, very close to a stable version we would want.

Next step is adding the code.

**Comment by smiths** [comment]
Thanks for the update @danscime.  I was able to view your Jupyter notebook from the Jupyter Lab web-page in surprisingly little time.   I do not have Jupyter lab installed, but the web interface was easy to open and import Projectile-Manual.ipynb.  Overall the documentation looks nice.  I don't know if you were looking for feedback, but I noticed a few small issues:

- In the Table of Symbols some of the superscripts are not shown properly (m/s2).
- the figures do not show up.
- the hibbeler citation is lower case (not a big deal, but it looks different from the usual)

I checked several cross-references in the document and they worked well.  I also like the appearance.  Neat and tidy.  The traceability matrices fit fairly well on a large screen, better than I expected.

**Comment by danscime** [comment]
@smiths Addressing your points:

1. I was testing differences between using HTML and embedded Latex to see the difference between the two. The Latex way would seem to keep everything more uniform and tidy. The end goal would be to have the math symbols generated in a similar way that the eventual MathJax would be generated (as in 479).

2. Projectile currently only has one figure. On my machine, it shows up. The image file needs to be in the same directory as the `.ipynb` file.
<img width="818" alt="Screen Shot 2019-07-24 at 2 15 46 PM" src="https://user-images.githubusercontent.com/28261339/61818096-0ead4880-ae1e-11e9-89bf-7d36fcc47976.png">

3. The citation appears nicely for me, so I'm not sure about that one.
<img width="660" alt="Screen Shot 2019-07-24 at 2 21 12 PM" src="https://user-images.githubusercontent.com/28261339/61818229-677ce100-ae1e-11e9-9bd9-bccd6de6c88f.png">

**Comment by smiths** [comment]
Sounds good @danscime.  The figure likely didn't appear because I used the web-interface to Jupyter.  I probably would have to upload it along with the .ipynb file.  The file is in your folder, so I'm sure this would work with a local Jupyter installation.

With respect to the citation, I didn't mean the reference, I meant the citation itself.  I close the Jupyter lab notebook, but it looked something like hibbeler2004 where it was cited.  I thought it would be Hibbeler2004.  Not a big deal

**Comment by danscime** [comment]
@smiths yes the citation is lowercase as in `hibbeler2004`. I think these lowercase references appear throughout all of the examples.

**Comment by smiths** [comment]
@danscime, I just verified that you are right about the citations being in lowercase.  I never noticed before.  😄 I do prefer the citations to start with an uppercase character, but given that I hadn't noticed before now, I guess I can't say that my preference is that strong.  😄 Down the road, it would be nice if the citation style was a variability we could modify.  Not something to worry about right now though.

**Comment by danscime** [comment]
Code was added to end of Projectile file in 211ce7ea575f6d354213c8943ee1aa1fb59c7020. This issue can maybe be closed for now or have a new issue opened for next steps.
